### PR TITLE
Fire Rune Rework

### PR DESCRIPTION
**Pull request by Shushishtok**
- Reworked Basic Fire rune (Fireball). Now applies an explosion on contact or at the expiration point which deals minor damage to enemies and sets them ablaze, causing them to take more damage over a short duration. 
    - Explosion particle has been reworked to get CP1 specifying its radius.
    - Radius has been increased from 125 to 200, and now increases more significantly. The radius no longer affects the projectile's radius, which made it almost never hit additional targets. Reworked damage slightly - most of the damage comes from the DoT now rather than the explosion itself.
- Added enums for generic status effects and generic effects - currently only holding the BURNING key for fire based effects.
- Replaced the naming in the runes localization files to use the weapon names enum and variables to prevent mistakes.

### CODE REVIEW

File: game/scripts/vscripts/weapons/basic_fire.ts [hunk 1]
```diff
@@ -21,7 +22,7 @@ export class basic_fire extends BaseWeapon {
 			ability: this,
 			startLoc: loc,
 			trailEffect: "particles/projectiles/fireball/fireball.vpcf",
-			radius: radius,
+			radius: 85,
```
Comment (Nibuja05) [inline game/scripts/vscripts/weapons/basic_fire.ts:25]: why do we have a constant radius here?

**Review by Nibuja05 — COMMENTED** [review]
Looking good (also ingame!) so far, just a small thing to do

File: game/scripts/vscripts/weapons/basic_fire.ts (hunk 1 above)
Comment (Shushishtok) [reply game/scripts/vscripts/weapons/basic_fire.ts:25]: Mentioned in the description. The issue with giving it the radius means that the projectile always came in contact of the unit at the edge of the AoE, meaning it never hit more than one target. Additionally, with high enough radius, it started hitting targets that visually it didn't actually touch.

Can make it a proper ability special value though.

**Comment by Shushishtok** [comment]
New commit. In this commit, I edited the localization to be ability-based (in the tooltip generator), and all runes are now working properly with it.

In addition, I also added sounds to Fireball.

**Review by Nibuja05 — APPROVED** [review]

**Pull request merged**